presto-main/src/main/resources/webapp/src/components/SessionProps.jsx

```diff
@@ -0,0 +1,203 @@
+    const sesstionProps = React.useRef({ data: [] });
+    const clickedRow = React.useRef([]);
+    const [filter, setFilter] = React.useState({ text: '', data: [] });
+    const filterData = (event) => {
+        const filteredData = sesstionProps.current.data.filter(
+            item => item[0].toLowerCase().includes(event.target.value.toLowerCase())
+        );
+        setFilter({ text: event.target.value, data: filteredData });
+    };
+    const clearFilter = () => {
+        if (filter.text) {
+            setFilter({ text: '', data: sesstionProps.current.data });
+        }
+    };
+
+    const internalChangeHandler = async (row, old) => {
+        // update the list to new value first, but verify the new value on the background
+        setFilter({ ...filter });
+        try {
+            await createClient().query(
+                `set session ${row[0]}=${row[3] === 'varchar' ? ("'" + row[1] + "'") : row[1]}`
+            );
+            if (changeHandler) {
+                changeHandler(row);
+            }
+        } catch (e) {
+            // revert the session back to previous value and update the list
+            row[1] = old;
+            setFilter({ ...filter });
+        }
+    };
+
+    const highlightAltered = [
+        {
+            when: row => row[1] !== row[2],
+            style: {
+                fontWeight: 'bold',
+                color: '#DFC7F2',
+            },
+        }
+    ];
+
+    const COLUMNS = [
+        {
+            name: 'Name',
+            selector: row => row[0],
+            sortable: true,
+            wrap: true,
+            minwidth: '350px',
+        },
+        {
+            name: 'Current Value',
+            selector: row => row[1],
+            cell: row => <EditableCell row={row} changeHandler={internalChangeHandler} />,
+            width: '150px',
+            wrap: true,
+        },
+        {
+            name: 'Default Value',
+            selector: row => row[2],
+            width: '150px',
+            wrap: true,
+        },
+        {
+            name: 'Date Type',
+            selector: row => row[3],
+            width: '100px',
+            wrap: true,
+        },
+        {
+            name: 'Description',
+            selector: row => row[4],
+            minwidth: '100px',
+            wrap: true,
+        },
+    ];
+
+    const rowClicked = (row) => {
+        if (clickedRow.current.length > 0 && clickedRow.current[0] !== row[0]) {
+            // change the previous one back to readOnly mode
+            clickedRow.current[5] = true;
+        }
+        // update readOnly as false to trigger the useEffect function of EditableCell
+        row[5] = false;
+        clickedRow.current = row;
+        setFilter({ ...filter });
+    };
+
+    React.useEffect(() => {
+        //fetch session properties:
+        async function getSessionProps() {
+            const client = createClient();
+            const prestoQuery = await client.query('show session');
```

Comment:
really cool stuff!

Reply:
this is all based on the `presto-js-client` from you